Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/parser/PRPAIN201306UV02Parser.java

```diff
@@ -132,15 +130,14 @@ public static String getReceiverHCID(PRPAIN201306UV02 response) {
                 .getValue().getId().get(0).getRoot();
         }
         //If representedOrganization Id root is null get id from device
-        if (NullChecker.isNullish(id)) {
-            if (response != null && response.getReceiver() != null && response.getReceiver().get(0) != null
-                && response.getReceiver().get(0).getDevice() != null
-                && response.getReceiver().get(0).getDevice().getId() != null
-                && response.getReceiver().get(0).getDevice().getId().get(0) != null
-                && response.getReceiver().get(0).getDevice().getId().get(0).getRoot() != null
-                && !response.getReceiver().get(0).getDevice().getId().get(0).getRoot().isEmpty()) {
-                id = response.getReceiver().get(0).getDevice().getId().get(0).getRoot();
-            }
+        if (NullChecker.isNullish(id) && response != null && response.getReceiver() != null
+            && response.getReceiver().get(0) != null
+            && response.getReceiver().get(0).getDevice() != null
+            && response.getReceiver().get(0).getDevice().getId() != null
+            && response.getReceiver().get(0).getDevice().getId().get(0) != null
+            && response.getReceiver().get(0).getDevice().getId().get(0).getRoot() != null
+            && !response.getReceiver().get(0).getDevice().getId().get(0).getRoot().isEmpty()) {
```

Comment:
You can combine the line 138 and 139 to the following: 
CollectionUtils.isNotEmpty(response.getReceiver().get(0).getDevice().getId().get(0).getRoot())

Reply:
CollectionUtils.isNotEmpty method accepts a collection as argument. In this case, response.getReceiver().get(0).getDevice().getId().get(0).getRoot() its a String.